### PR TITLE
Prospector

### PR DESCRIPTION
**Pull request by jeamland**
Switch from flake8 to prospector for linting.

### CODE REVIEW

File: wsproto/connection.py [hunk 1]
```diff
@@ -393,6 +393,34 @@ def _process_connection_request(self, event):
 
         return ConnectionRequested(proposed_subprotocols, event)
 
+    def extension_accept(self, extensions_header):
```
Comment (Kriechi) [inline wsproto/connection.py]: Should this be public API?
It doesn't really do anything with the connection, and it is only called via `accept`.

Comment (jeamland) [inline wsproto/connection.py]: Are you suggesting it should be renamed to _extension_accept or that it should go back into the function I pulled it out of?

I'm fine with the former.

Comment (Kriechi) [inline wsproto/connection.py]: `_extension_accept` to mark it as *private* function.

**Comment by Kriechi** [comment]
General 👍 on the code changes and fixing dormant bugs (I'm looking at you *default `[]`* parameter!)

I'm neutral about prospector vs. flake8. 
My 2 cents: All projects under the python-hyper umbrella should have a somewhat similar strategy for testing + linting + travis matrix etc.

**Pull request merged**